### PR TITLE
avoid regex in navigate module

### PR DESCRIPTION
**Pull request by karussell**

### CODE REVIEW

File: navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java [hunk 1]
```diff
@@ -188,13 +188,11 @@ private GHResponse calcRoute(List<Double> favoredHeadings, List<GHPoint> request
      * The url looks like: ".../{profile}/1.522438,42.504606;1.527209,42.504776;1.526113,42.505144;1.527218,42.50529?.."
      */
     private List<GHPoint> getPointsFromRequest(HttpServletRequest httpServletRequest, String profile) {
-
         String url = httpServletRequest.getRequestURI();
-        url = url.replaceFirst("/navigate/directions/v5/gh/" + profile + "/", "");
-        url = url.replaceAll("\\?[*]", "");
```
Comment (karussell) [inline navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java:194]: This is unnecessary as requestURI does not contain query string or `?`

File: navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java [hunk 2]
```diff
@@ -188,13 +188,11 @@ private GHResponse calcRoute(List<Double> favoredHeadings, List<GHPoint> request
      * The url looks like: ".../{profile}/1.522438,42.504606;1.527209,42.504776;1.526113,42.505144;1.527218,42.50529?.."
      */
     private List<GHPoint> getPointsFromRequest(HttpServletRequest httpServletRequest, String profile) {
-
         String url = httpServletRequest.getRequestURI();
-        url = url.replaceFirst("/navigate/directions/v5/gh/" + profile + "/", "");
```
Comment (karussell) [inline navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java:193]: Avoid regex for user input

File: navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java [hunk 3]
```diff
@@ -188,13 +188,11 @@ private GHResponse calcRoute(List<Double> favoredHeadings, List<GHPoint> request
...
-        url = url.replaceAll("\\?[*]", "");
-
+        String urlStart = "/navigate/directions/v5/gh/" + profile + "/";
+        if (url.indexOf(urlStart) != 0) throw new IllegalArgumentException("Incorrect URL " + url);
```
Comment (easbar) [inline navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java:193]: Maybe rather use `!url.startsWith("/navigate...")`?

Comment (karussell) [reply navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java:193]: Ah, yeah, probably just `equals` is best.

**Pull request merged**